binaries/cuprated/src/rpc/handler.rs

```diff
@@ -0,0 +1,102 @@
+//! Dummy implementation of [`RpcHandler`].
+
+use std::task::{Context, Poll};
+
+use cuprate_rpc_types::{
+    bin::{BinRequest, BinResponse},
+    json::{JsonRpcRequest, JsonRpcResponse},
+    other::{OtherRequest, OtherResponse},
+};
+use futures::{channel::oneshot::channel, future::BoxFuture};
+use serde::{Deserialize, Serialize};
+use tower::Service;
+
+use cuprate_blockchain::service::BlockchainReadHandle;
+use cuprate_helper::asynch::InfallibleOneshotReceiver;
+use cuprate_json_rpc::Id;
+use cuprate_rpc_interface::{RpcError, RpcHandler};
+use cuprate_txpool::service::TxpoolReadHandle;
+
+use crate::rpc::{bin, json, other};
+
+/// TODO
+#[derive(Clone)]
+pub struct CupratedRpcHandler {
+    /// Should this RPC server be [restricted](RpcHandler::restricted)?
+    //
+    // INVARIANT:
+    // We don't need to include this in `state` and check for
+    // `self.is_restricted()` because `cuprate-rpc-interface` handles that.
+    pub restricted: bool,
+
+    /// State needed for request -> response mapping.
+    pub state: CupratedRpcHandlerState,
+}
+
+/// TODO
+#[derive(Clone)]
+pub struct CupratedRpcHandlerState {
+    /// Read handle to the blockchain database.
+    pub blockchain: BlockchainReadHandle,
+
+    /// Read handle to the transaction pool database.
+    pub txpool: TxpoolReadHandle,
+}
```

Comment:
This is `cuprated`'s `RpcHandler`.

I separated the state into another struct so the handler only needs to clone that instead of all of `CupratedRpcHandler`. It only save a single `bool` for now though.